src/core/Akka.Cluster/Cluster.cs

```diff
@@ -264,7 +264,7 @@ public void Join(Address address)
         /// <returns>Task which completes, once current cluster node reaches <see cref="MemberStatus.Up"/> state.</returns>
         public Task JoinAsync(Address address, CancellationToken token = default)
         {
-            if (_isTerminated)
+            if (_isTerminated.Value)
```

Comment:
This can be avoided if implicit casting of AtomicBool -> bool is allowed. Would minimize the amount of work we need to do to fix this API breaking change

Reply:
(the work is already done though)